hazelcast/src/main/java/com/hazelcast/internal/partition/impl/InternalPartitionServiceImpl.java

```diff
@@ -832,15 +833,17 @@ public boolean isPartitionReplicaVersionStale(int partitionId, long[] versions,
 
     @Override
     public Map<Address, List<Integer>> getMemberPartitionsMap() {
-        final Collection<Member> dataMembers = node.getClusterService().getMembers(DATA_MEMBER_SELECTOR);
-        final int dataMembersSize = dataMembers.size();
+        Collection<Member> dataMembers = node.getClusterService().getMembers(DATA_MEMBER_SELECTOR);
+        int dataMembersSize = dataMembers.size();
```

Comment:
Good finding! Actually this can happen if you have a LiteMember only cluster. There is even a test for it, which didn't fail since an exception is thrown in that case. So effectively nothing would have happened, since the exception is thrown before the `ArrayList` is created. But that's risky and so a fix is the better solution: https://github.com/hazelcast/hazelcast/pull/8054